feat: aggregated provider states

YOU54F opened the pull request:
Provide an endpoint to retrieve provider states to aid discoverability.

Initial work performed in https://github.com/pact-foundation/pact_broker/tree/feat/aggregated-provider-states by @bethesque 

Allowed methods: `GET`

Path: `/pacts/provider/{provider}/provider-states`

This resource returns a aggregated de-duplicated list of all provider states for a given provider.

Provider states are collected from the latest pact on the main branch for any dependant consumers.

Example response

```json
{
    "providerStates": [
        {
            "name": "an error occurs retrieving an alligator"
        },
        {
            "name": "there is an alligator named Mary"
        },
        {
            "name": "there is not an alligator named Mary"
        }
    ]
}
```

pahnin commented on lib/pact_broker/pacts/content.rb:
```diff
@@ -116,12 +131,14 @@ def interactions
+        end
       end
 
       def messages_or_interaction_or_empty_array
```
I would change this to 

`def messages_and_or_interaction_or_empty_array`

YOU54F replied:
done :)

YOU54F committed:
feat: aggregated provider state endpoint
# Provider States - Aggregated view by provider

Allowed methods: `GET`

Path: `/pacts/provider/{provider}/provider-states`

This resource returns a aggregated de-duplicated list of all provider states for a given provider.

Provider states are collected from the latest pact on the main branch for any dependant consumers.

Example response

```json
{
    "providerStates": [
        {
            "name": "an error occurs retrieving an alligator"
        },
        {
            "name": "there is an alligator named Mary"
        },
        {
            "name": "there is not an alligator named Mary"
        }
    ]
}
```
